src/olmo_core/data/source_mixture.py

```diff
@@ -180,6 +180,10 @@ class SourceMixtureDatasetConfig(Config):
     """
     The data type of the dataset.
     """
+    global_batch_size: int
```

Comment:
nit: It would be nice to not have to define the global_batch_size at two different places in the config (here and in the data loader). But I took a quick look at how that could be done and it doesn't look super straightforward/clean to communicate this value from the dataloader to here.